lib/ecto/schema.ex

```diff
@@ -472,22 +483,23 @@ defmodule Ecto.Schema do
   to `:binary_id` but such can be configured with the
   `@primary_key` attribute.
   """
-  defmacro embedded_schema([do: block]) do
-    schema(__CALLER__, nil, false, :binary_id, block)
+  defmacro embedded_schema(opts \\ [], do: block) do
+    schema(__CALLER__, nil, false, :binary_id, block, opts)
   end
-
   @doc """
   Defines a schema struct with a source name and field definitions.
 
   An additional field called `__meta__` is added to the struct for storing
   internal Ecto state. This field always has a `Ecto.Schema.Metadata` struct
   as value and can be manipulated with the `Ecto.put_meta/2` function.
   """
-  defmacro schema(source, [do: block]) do
-    schema(__CALLER__, source, true, :id, block)
+  defmacro schema(source, opts \\ [], [do: block]) do
```

Comment:
Oops... I forgot we do not accept any options today. What if instead of an option we use a module attribute then? `@ecto_derive_inspect_for_redacted_fields` which defaults to true? Sorry for misleading...